src/app/drag-drop.service.ts

```diff
@@ -0,0 +1,80 @@
+import { Injectable, EventEmitter } from '@angular/core';
+
+@Injectable()
+export class DragDropService {
+
+  private dragData: any;
+  private dropContainer: any;
+  isDragging: boolean;
+  dropPanelId: any;
+  dragStart: any;
+  dragStyle: any;
+  dragPlaceHolderHTML: any;
+
+  componentDroppedOutsidePanel: EventEmitter<any>;
+  componentDroppedInsidePanel: EventEmitter<any>;
+
+  constructor() {
+    this.componentDroppedOutsidePanel = new EventEmitter();
+    this.componentDroppedInsidePanel = new EventEmitter();
+    this.isDragging = false;
+  }
+
+  setDragStart(x, y, offsetX, offsetY, target) {
+
+    let rect = target.getBoundingClientRect();
+    let height = rect.height;
+    let width = rect.width;
+
+    this.dragStart = {x, y, height, width, offsetX, offsetY};
+    this.dragPlaceHolderHTML = target.innerHTML;
+  }
+
+  getDraggingHeaderWidth() {
+    return this.dragStart.width;
+  }
+
```

Comment:
This will be used when dragging a component header. When we are hovering a drop zone it will use this value to transform the x position of the header components.